controllers/storageclass/storageclass_controller.go

```diff
@@ -192,6 +187,28 @@ func (r *StorageClassWatcher) getFlashSystemClusterByStorageClass(sc *storagev1.
 	return foundCluster, fmt.Errorf(msg)
 }
 
+func (r *StorageClassWatcher) getSecret(sc *storagev1.StorageClass) (corev1.Secret, error) {
+	secret := &corev1.Secret{}
+	secretName, secretNamespace := sc.Parameters[util.DefaultSecretNameKey], sc.Parameters[util.DefaultSecretNamespaceKey]
+	if secretName == "" || secretNamespace == "" {
+		secretName, secretNamespace = sc.Parameters[util.ProvisionerSecretNameKey], sc.Parameters[util.ProvisionerSecretNamespaceKey]
+		if secretName == "" || secretNamespace == "" {
+			r.Log.Error(nil, "failed to find secret name or namespace in StorageClass")
+			return *secret, fmt.Errorf("failed to find secret name or namespace in StorageClass")
```

Comment:
please extract the error message into variable and use it for both `Log.Error` and `fmt.Error`